astropy/utils/decorators.py

```diff
@@ -751,3 +752,230 @@ def _get_function_args(func, exclude_args=()):
                 all_args[arg_type] = None
 
     return all_args
+
+
+def format_doc(docstring, *args, **kwargs):
+    """
+    Replaces the docstring of the decorated object and then formats it.
+
+    This decorator keeps the objects signature and only alters the saved
+    ``__doc__`` property (so it cannot be used as class decorator in python2).
+    The formatting works like :meth:`str.format` and if the decorated object
+    already has a docstring this docstring can be included in the new
+    documentation if you use the ``{__doc__}`` placeholder.
+    It's primary use is if multiple functions have the same or only
+    a slightly different *long* docstring and you want to DRY
```

Comment:
"It's primary use is if multiple functions have the same or only a slightly different *long* docstring ->

"Its primary use is for reusing a *long* docstring in multiple functions when it is the same or only slightly different between them."  (drop the part about DRY)